### PR TITLE
Add support for automatic tagging (:1, :1.2, :1.2.3, etc) for semantic versions

### PR DESCRIPTION
**Pull request by kylelemons**
# Enabling the feature

This feature is off by default.  To enable it, either:

* `PLUGIN_AUTO_TAG` for drone plugins
* `--auto_tag` for command-line use

This changes the way tags (both `--tags` and `PLUGIN_TAGS`) are interpreted.  Now, instead of saying `1,1.2,1.2.3,latest` you can say `v1.2.3,latest`

This mostly follows semver.org, except that it also swaps underscores (which are not permitted) for hyphens, as otherwise it will likely be confusing when you try `v1.2.3+linux_arm` and your semver tagging breaks.

# Semantics

This parses all tags as semantic versions (after performing a swap of `_` for `-` as required by semver.org).  If they parse correctly, the following logic happens:

* If the semver is pre-release, it is emitted directly (e.g. `v1.2.3-alpha1` tags `1.2.3-alpha1`)
* If the semver has build info, it is appended to all tags (e.g. `v1.2.3+linux` tags `1+linux`, `1.2+linux`, etc)
* If it is not a valid semver, nothing changes

# Alternatives

Instead of interpreting PLUGIN_TAGS differently, we could make PLUGIN_AUTO_TAGS a list that would be required to parse correctly as semantic versions, which might avoid some classes of error.  That would make invocation look like

```
-e PLUGIN_TAGS=latest
-e PLUGIN_AUTO_TAGS=$DRONE_TAG
```

This probably works fine for most drone workflows, though it will break workflows where e.g. someone tags a build `for_testing` or something along those lines, so I did not go with this approach initially.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,3 +51,23 @@ docker run --rm \
...
     plugins/kaniko:linux-amd64
 ```
+
+### Automatic Tagging
+
+With auto tagging enabled, semantic versions can be passed to PLUGIN_TAGS directly for expansion:
+
+```console
+docker run --rm \
+    -e PLUGIN_TAGS=v1.2.3,latest \
```
Comment (shubham149) [inline README.md:61]: Add `-e PLUGIN_AUTO_TAG=true`

File: cmd/kaniko-docker/main.go [hunk 2]
```diff
@@ -61,6 +61,11 @@ func main() {
 			EnvVar:   "PLUGIN_TAGS",
 			FilePath: ".tags",
 		},
+		cli.BoolFlag{
```
Comment (shubham149) [inline cmd/kaniko-docker/main.go:64]: Add this flag for kaniko-gcr & kaniko-ecr as well.

**Review by shubham149 — COMMENTED** [review]
Sorry for late review. I missed the notification for this pr review.

File: cmd/kaniko-docker/main.go (hunk 2 above)
Comment (kylelemons) [reply cmd/kaniko-docker/main.go:67]: Done

**Review by shubham149 — APPROVED** [review]

**Comment by shubham149** [comment]
LGTM. We can merge after the build check passes.

**Pull request merged**